customcommands/interval_test.go

```diff
@@ -1,315 +1,335 @@
+			LastRun:  null.TimeFrom(zero),
+			RefTime:  zero,
+			Want:     runOnDate(day(2))},
+
+		{Name: "5m interval excluding days and hours",
+			Interval: 5,
+			Exclude:  exclude{Days: []time.Weekday{time.Tuesday}, Hours: allHoursExcept(1)},
+			LastRun:  null.TimeFrom(zero.Add(2 * time.Hour)),
+			RefTime:  zero.Add(2 * time.Hour),
+			Want:     runOnDate(day(3), hour(1), minute(0))},
+
+		{Name: "exclude all hours",
+			Interval: 5,
+			Exclude:  exclude{Hours: allHours()},
+			LastRun:  null.Time{},
+			RefTime:  now,
+			Want:     neverRuns()},
+
+		{Name: "exclude all days",
+			Interval: 5,
+			Exclude:  exclude{Days: allDays()},
+			LastRun:  null.Time{},
+			RefTime:  now,
+			Want:     neverRuns()},
+
+		{Name: "interval <5m never runs",
+			Interval: 1,
+			LastRun:  null.Time{},
+			RefTime:  now,
+			Want:     neverRuns()},
+
+		{Name: "interval >31d never runs",
+			Interval: 31*24*60 + 1,
+			LastRun:  null.Time{},
+			RefTime:  now,
+			Want:     neverRuns()},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.Name, func(t *testing.T) {
+			days := make([]int64, len(tt.Exclude.Days))
+			for i, v := range tt.Exclude.Days {
+				days[i] = int64(v)
+			}
+			cmd := &models.CustomCommand{
+				TriggerType:               int(CommandTriggerInterval),
+				TimeTriggerInterval:       tt.Interval,
+				TimeTriggerExcludingDays:  days,
+				TimeTriggerExcludingHours: tt.Exclude.Hours,
+				LastRun:                   tt.LastRun,
+			}
+			got := CalcNextRunTime(cmd, tt.RefTime).UTC()
+
+			testDesc := fmt.Sprintf("interval every %d mins\n\texclude hours: %v\n\texclude days: %v",
+				tt.Interval, tt.Exclude.Hours, tt.Exclude.Days)
+			testHelper(t, testDesc, got, tt.Want, tt.RefTime)
+		})
 	}
+}
 
-	cc.TextTrigger = "0 * * * 2"
-
-	nextRun = CalcNextRunTime(cc, tim)
-	expected = tim.UTC().Add(time.Hour * 24)
-
-	if nextRun != expected {
-		t.Error("incorrect next run, should be: ", expected, ", got: ", nextRun)
+func TestCronNextRunTime(t *testing.T) {
+	var now = time.Now().UTC()
+	var zero = time.Time{}.UTC()
+
+	tests := []struct {
+		Name    string
+		Cron    string
+		Exclude exclude
+		RefTime time.Time
+
+		Want expectedTime
+	}{
+		{Name: "basic on minute 5",
+			Cron:    "5 * * * *",
+			RefTime: zero.Add(3 * time.Minute),
+			Want:    runOnDate(minute(5))},
+
+		{Name: "on minute 0 of day 2",
+			Cron:    "0 * 2 * *",
+			RefTime: zero,
+			Want:    runOnDate(day(2), hour(0), minute(0))},
+
+		{Name: "on minute 0 of feb",
+			Cron:    "0 * * 2 *",
+			RefTime: zero,
+			Want:    runOnDate(month(time.February), day(1), hour(0), minute(0))},
+
+		{Name: "on minute 0 of tues",
+			Cron:    "0 * * * 2",
+			RefTime: zero,
+			Want:    runOnDate(day(2), hour(0), minute(0))},
+
+		{Name: "exclude all hours",
+			Cron:    "* * * * *",
+			Exclude: exclude{Hours: allHours()},
+			RefTime: now,
+			Want:    neverRuns()},
+
+		{Name: "exclude hour 0 from cron running on hour 0",
+			Cron:    "* 0 * * * *",
```

Comment:
Extra param here?

Reply:
Good catch. I will correct this and add some validation to ensure we don't test invalid crons shortly.